### ISSUE TITLE
using Catlab fails

### ISSUE THREAD

**Issue by djinnome** [AlgebraicJulia/Catlab.jl#618]
```julia
$ julia
 julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.7.2 (2022-02-06)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |
julia> using Catlab
[ Info: Precompiling Catlab [134e5e36-593f-5add-ad60-77f754baafbe]
ERROR: LoadError: MethodError: no method matching names(::Base.Broadcast.BroadcastFunction{Module})
Closest candidates are:
  names(::Module; all, imported) at /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/base/reflection.jl:102
Stacktrace:
  [1] top-level scope
    @ ~/.julia/packages/Catlab/2P2DD/src/categorical_algebra/CSetDataStructures.jl:14
  [2] include(mod::Module, _path::String)
    @ Base ./Base.jl:418
  [3] include(x::String)
    @ Catlab ~/.julia/packages/Catlab/2P2DD/src/Catlab.jl:1
  [4] top-level scope
    @ ~/.julia/packages/Catlab/2P2DD/src/Catlab.jl:8
  [5] include
    @ ./Base.jl:418 [inlined]
  [6] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1318
  [7] top-level scope
    @ none:1
  [8] eval
    @ ./boot.jl:373 [inlined]
  [9] eval(x::Expr)
    @ Base.MainInclude ./client.jl:453
 [10] top-level scope
    @ none:1
in expression starting at /Users/zuck016/.julia/packages/Catlab/2P2DD/src/categorical_algebra/CSetDataStructures.jl:1
in expression starting at /Users/zuck016/.julia/packages/Catlab/2P2DD/src/Catlab.jl:1
ERROR: Failed to precompile Catlab [134e5e36-593f-5add-ad60-77f754baafbe] to /Users/zuck016/.julia/compiled/v1.7/Catlab/jl_9e64Ab.
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IO, internal_stdout::IO, ignore_loaded_modules::Bool)
   @ Base ./loading.jl:1466
 [3] compilecache(pkg::Base.PkgId, path::String)
   @ Base ./loading.jl:1410
 [4] _require(pkg::Base.PkgId)
   @ Base ./loading.jl:1120
 [5] require(uuidkey::Base.PkgId)
   @ Base ./loading.jl:1013
 [6] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:997
```

**Comment by epatters** [comment]
That's odd. I can import and run Catlab (both v0.13.8 and master) on Julia v1.7.2.

Can you provide more information? What OS and what version of Catlab are you using?

**Comment by epatters** [comment]
A similar error was reported in #509, but I thought it was fixed.

**Comment by djinnome** [comment]
MacOS Catalina 10.15.7

I tried it with two version of Catlab.   The most recent version, and the version specified in https://github.com/AlgebraicJulia/Structured-Epidemic-Modeling/blob/723cf5d20ae60bc908f24c35eb5d195cef9ffa8a/Project.toml#L4

**Comment by djinnome** [comment]
I should disclose that I am very new to Julia, so don't rule out the possibility that I did something dumb.
For example, when I first added Catlab, it failed to build, but now I can't get that error message, even after doing rm Catlab followed by add Catlab.

**Comment by djinnome** [comment]
One thing I did do, was install Catlab *after* I installed Omega.  
If I do `rm --all`, then `add Catlab`, followed by `add Omega`, then the error shows up:

```julia
(@v1.7) pkg> add Omega
   Resolving package versions...
    Updating `~/.julia/environments/v1.7/Project.toml`
  [134e5e36] ↓ Catlab v0.13.8 ⇒ v0.13.1
  [1af16e33] + Omega v0.1.1
    Updating `~/.julia/environments/v1.7/Manifest.toml`
  [621f4979] + AbstractFFTs v1.1.0
  [1520ce14] + AbstractTrees v0.2.1
  [79e6a3ab] + Adapt v1.1.0
  [dce04be8] + ArgCheck v2.3.0
  [4fba245c] + ArrayInterface v5.0.7
  [4c555306] + ArrayLayouts v0.1.5
  [6e4b80f9] + BenchmarkTools v0.7.0
  [b99e7846] + BinaryProvider v0.5.10
  [db1e321a] + Callbacks v0.1.0
  [7057c7e9] + Cassette v0.3.10
  [134e5e36] ↓ Catlab v0.13.8 ⇒ v0.13.1
  [d360d2e6] + ChainRulesCore v1.14.0
  [9e997f8a] + ChangesOfVariables v0.1.2
  [523fee87] + CodecBzip2 v0.7.2
  [944b1d66] + CodecZlib v0.6.0
  [3da002f7] ↓ ColorTypes v0.11.0 ⇒ v0.8.1
  [5ae59095] ↓ Colors v0.12.8 ⇒ v0.9.6
  [bbf7d656] + CommonSubexpressions v0.3.0
  [85a47980] + Dictionaries v0.3.20
  [163ba53b] + DiffResults v1.0.3
  [b552c78f] + DiffRules v1.10.0
  [31c24e10] + Distributions v0.23.8
  [ffbed154] + DocStringExtensions v0.8.6
  [bbc10e6e] + DynamicHMC v2.2.0
  [7a1cc6ca] + FFTW v1.4.6
  [1a297f60] + FillArrays v0.8.14
  [6a86dc24] + FiniteDiff v2.11.0
  [53c48c17] ↓ FixedPointNumbers v0.8.4 ⇒ v0.6.1
  [587475ba] + Flux v0.8.3
  [f6369f11] + ForwardDiff v0.10.25
  [069b7b12] + FunctionWrappers v1.1.2
  [7869d1d1] + IRTools v0.3.3
  [615f187c] + IfElse v0.1.1
  [313cdc1a] + Indexing v1.1.1
  [3587e190] + InverseFunctions v0.1.3
  [92d709cd] + IrrationalConstants v0.1.1
  [e5e0dc1b] + Juno v0.7.2
  [496038af] + Lens v0.1.1
  [d3d80556] + LineSearches v7.1.1
  [6fdf6af0] + LogDensityProblems v0.10.6
  [2ab3a3ac] + LogExpFunctions v0.3.12
  [1914dd2f] + MacroTools v0.5.9
  [b8f27783] + MathOptInterface v0.10.9
  [fdba3010] + MathProgBase v0.7.8
  [e89f7d12] + Media v0.5.0
  [e1d29d7a] + Missings v1.0.2
  [d8a4904e] + MutableArithmetics v0.3.3
  [d41bc354] + NLSolversBase v7.8.2
  [76087f3c] + NLopt v0.6.5
  [872c559c] + NNlib v0.6.6
  [77ba4419] + NaNMath v0.3.7
  [1af16e33] + Omega v0.1.1
  [429524aa] + Optim v1.6.2
  [90014a1f] + PDMats v0.10.1
  [d96e819e] + Parameters v0.12.3
  [85a6dd25] + PositiveFactorizations v0.2.4
  [92933f4c] + ProgressMeter v1.7.2
  [1fd47b50] + QuadGK v2.4.2
  [189a3867] ↓ Reexport v1.2.2 ⇒ v0.2.0
  [79098fc4] + Rmath v0.7.0
  [a2af1166] + SortingAlgorithms v1.0.1
  [b8ccf107] + Spec v0.2.0
  [276daf66] + SpecialFunctions v0.10.3
  [03a91e81] + SplitApplyCombine v1.2.1
  [aedffcd0] + Static v0.6.0
  [82ae8749] + StatsAPI v1.2.2
  [2913bbd2] + StatsBase v0.33.16
  [4c63d2b9] + StatsFuns v0.9.7
  [9f7883ad] + Tracker v0.2.19
  [3bb67fe8] + TranscodingStreams v0.9.6
  [84d833dd] + TransformVariables v0.3.12
  [9d95f2ec] + TypedTables v1.4.0
  [3a884ed6] + UnPack v1.0.2
  [b8865327] + UnicodePlots v1.4.0
  [a5390f91] + ZipFile v0.8.4
  [e88e6eb3] + Zygote v0.4.12
  [700de1a5] + ZygoteRules v0.2.2
  [6e34b625] + Bzip2_jll v1.0.8+0
  [f5851436] + FFTW_jll v3.3.10+0
  [1d5cc7b8] + IntelOpenMP_jll v2018.0.3+2
  [856f044c] + MKL_jll v2022.0.0+0
  [079eb43e] + NLopt_jll v2.7.1+0
  [efe28fd5] + OpenSpecFun_jll v0.5.5+0
  [f50d1b31] + Rmath_jll v0.3.0+0
  [4af54fe1] + LazyArtifacts
  [9abbd945] + Profile
  [4607b0f0] + SuiteSparse
Precompiling project...
  ✓ FixedPointNumbers
  ✓ Reexport
  ✓ PrettyTables
  ✓ ColorTypes
  ✓ Colors
  ✓ Compose
  ✗ Catlab
  10 dependencies successfully precompiled in 33 seconds (108 already precompiled)
  6 dependencies precompiled but different versions are currently loaded. Restart julia to access the new versions
  1 dependency errored. To see a full report either run `import Pkg; Pkg.precompile()` or load the package
```

I get the same error if I do it in the opposite order:

```julia
rm --all
add Omega
(@v1.7) pkg> add Catlab
   Resolving package versions...
    Updating `~/.julia/environments/v1.7/Project.toml`
  [134e5e36] + Catlab v0.13.1
    Updating `~/.julia/environments/v1.7/Manifest.toml`
  [15f4f7f2] + AutoHashEquals v0.2.0
  [134e5e36] + Catlab v0.13.1
  [a81c6b42] + Compose v0.9.3
  [e2d170a0] + DataValueInterfaces v1.0.0
  [85a47980] + Dictionaries v0.3.20
  [59287772] + Formatting v0.4.2
  [069b7b12] + FunctionWrappers v1.1.2
  [6b9d7cbe] + GeneralizedGenerated v0.3.3
  [313cdc1a] + Indexing v1.1.1
  [c8e1da08] + IterTools v1.4.0
  [82899510] + IteratorInterfaceExtensions v1.0.0
  [b14d175d] + JuliaVariables v0.2.4
  [9c8b4983] + LightXML v0.9.0
  [d8e11817] + MLStyle v0.4.10
  [442fdcdd] + Measures v0.3.1
  [71a1bf82] + NameResolution v0.1.5
  [8162dcfd] + PrettyPrint v0.2.0
  [08abe8d2] + PrettyTables v1.3.1
  [03a91e81] + SplitApplyCombine v1.2.1
  [3783bdb8] + TableTraits v1.0.1
  [bd369af6] + Tables v1.7.0
  [9d95f2ec] + TypedTables v1.4.0
  [94ce4f54] + Libiconv_jll v1.16.1+1
  [02c8fc9c] + XML2_jll v2.9.12+0
Precompiling project...
  ✗ Catlab
  0 dependencies successfully precompiled in 15 seconds (118 already precompiled)
  1 dependency errored. To see a full report either run `import Pkg; Pkg.precompile()` or load the package

(@v1.7) pkg> 
```

**Comment by djinnome** [comment]
Here is the full report:

```julia
julia> import Pkg

julia> Pkg.precompile()
Precompiling project...
  ✗ Catlab
  0 dependencies successfully precompiled in 15 seconds (118 already precompiled)

ERROR: The following 1 direct dependency failed to precompile:

Catlab [134e5e36-593f-5add-ad60-77f754baafbe]

Failed to precompile Catlab [134e5e36-593f-5add-ad60-77f754baafbe] to /Users/zuck016/.julia/compiled/v1.7/Catlab/jl_JBqGv6.
ERROR: LoadError: MethodError: no method matching names(::Base.Broadcast.BroadcastFunction{Module})
Closest candidates are:
  names(::Module; all, imported) at /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/base/reflection.jl:102
Stacktrace:
  [1] top-level scope
    @ ~/.julia/packages/Catlab/2P2DD/src/categorical_algebra/CSetDataStructures.jl:14
  [2] include(mod::Module, _path::String)
    @ Base ./Base.jl:418
  [3] include(x::String)
    @ Catlab ~/.julia/packages/Catlab/2P2DD/src/Catlab.jl:1
  [4] top-level scope
    @ ~/.julia/packages/Catlab/2P2DD/src/Catlab.jl:8
  [5] include
    @ ./Base.jl:418 [inlined]
  [6] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
    @ Base ./loading.jl:1318
  [7] top-level scope
    @ none:1
  [8] eval
    @ ./boot.jl:373 [inlined]
  [9] eval(x::Expr)
    @ Base.MainInclude ./client.jl:453
 [10] top-level scope
    @ none:1
in expression starting at /Users/zuck016/.julia/packages/Catlab/2P2DD/src/categorical_algebra/CSetDataStructures.jl:1
in expression starting at /Users/zuck016/.julia/packages/Catlab/2P2DD/src/Catlab.jl:1
Stacktrace:
 [1] pkgerror(msg::String)
   @ Pkg.Types /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/Pkg/src/Types.jl:68
 [2] precompile(ctx::Pkg.Types.Context; internal_call::Bool, strict::Bool, warn_loaded::Bool, already_instantiated::Bool, kwargs::Base.Pairs{Symbol, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Pkg.API /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/Pkg/src/API.jl:1362
 [3] precompile
   @ /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/Pkg/src/API.jl:1013 [inlined]
 [4] #precompile#220
   @ /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/Pkg/src/API.jl:1011 [inlined]
 [5] precompile()
   @ Pkg.API /Applications/Julia-1.7.app/Contents/Resources/julia/share/julia/stdlib/v1.7/Pkg/src/API.jl:1011
 [6] top-level scope
   @ REPL[20]:1

julia> 
```

**Comment by djinnome** [comment]
I guess the solution is to not install Omega, I'll bug Zenna with this error.

**Comment by epatters** [comment]
Here's what I think is going on. As we discovered in #509, old versions (< v1.0) of the package Reexport don't work on recent versions of Julia. So, starting in Catlab v0.13.2, we required Reexport v1.0+.

It looks like Omega or some package it depends on is forcing a downgrade of Reexport to v0.2 (which also forces a downgrade of Catlab to v0.13.1). So unfortunately I don't think there's much we can do about it.
